### PR TITLE
Replace failure crate with anyhow+thiserror

### PR DESCRIPTION
**Pull request by steveej**
Failure was [deprecated][0] and suggests anyhow+thiserror to replace
itself in dependent projects.

[0]: https://github.com/rust-lang-nursery/failure/pull/347

### CODE REVIEW

**Comment by steveej** [comment]
CI flake, Images failed to build.

/retest

**Review by cgwalters — APPROVED** [review]

**Comment by steveej** [comment]
> CI flake, Images failed to build.

I was mislead by this message:

```
   * could not run steps: step e2e-test failed: could not wait for build: the build e2e-test failed with reason DockerBuildFailed: Docker build strategy has failed. 
```

when in fact it was an unresolved import in the e2e tests which I missed when testing locally.

File: cincinnati/src/plugins/mod.rs [hunk 1]
```diff
@@ -14,7 +14,7 @@ use crate as cincinnati;
 use self::cincinnati::plugins::interface::{PluginError, PluginExchange};
 
 use async_trait::async_trait;
-use failure::{Error, Fallible, ResultExt};
+pub use commons::prelude_errors::*;
```
Comment (LalatenduMohanty) [inline cincinnati/src/plugins/mod.rs:17]: I have not seen this before. Can you please explain purpose of this?

File: cincinnati/src/lib.rs [hunk 2]
```diff
@@ -154,25 +151,27 @@ pub struct Empty;
 
 /// Errors that can be returned by the methods in this library
 pub mod errors {
+    use commons::prelude_errors::*;
```
Comment (LalatenduMohanty) [inline cincinnati/src/lib.rs:154]: I have not seen this before. Can you please explain purpose of this?

Comment (LalatenduMohanty) [reply cincinnati/src/lib.rs:154]: Also did not find any pointers in internet e.g. https://doc.rust-lang.org/std/prelude/index.html

Comment (steveej) [reply cincinnati/src/lib.rs:154]: `prelude` is a naming convention for modules which export commonly used items. AFAIU there's nothing right or wrong about it, and I decided to include `errors` in the name as this prelude is about error related items only.

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Pull request merged**